--- test/eqp.test
+++ test/eqp.test
@@ -521,13 +521,18 @@
 # documentation page eqp.html works. The C code is duplicated in test1.c
 # and wrapped in Tcl command [print_explain_query_plan] 
 #
+db close
+set cryptokey ""
+if {[sqlite3 -has-codec]} {
+  set cryptokey "-key xyzzy"
+}
 set boilerplate {
   proc explain_query_plan {db sql} {
     set stmt [sqlite3_prepare_v2 db $sql -1 DUMMY]
     print_explain_query_plan $stmt
     sqlite3_finalize $stmt
   }
-  sqlite3 db test.db
+  sqlite3 db test.db %KEY%
   explain_query_plan db {%SQL%}
   db close
   exit
@@ -536,7 +541,7 @@
 # Do a "Print Explain Query Plan" test.
 proc do_peqp_test {tn sql res} {
   set fd [open script.tcl w]
-  puts $fd [string map [list %SQL% $sql] $::boilerplate]
+  puts $fd [string map [list %SQL% $sql %KEY% $::cryptokey] $::boilerplate]
   close $fd
 
   uplevel do_test $tn [list {
@@ -556,6 +561,7 @@
 0 0 0 COMPOUND SUBQUERIES 1 AND 2 (EXCEPT)
 }]
 
+sqlite3 db test.db
 #-------------------------------------------------------------------------
 # The following tests - eqp-7.* - test that queries that use the OP_Count
 # optimization return something sensible with EQP.
