diff --git a/libevent/include/event2/event-config.h b/libevent/include/event2/event-config.h
new file mode 120000
--- /dev/null
+++ b/libevent/include/event2/event-config.h
@@ -0,0 +1,1 @@
+event-config.h.linux
\ No newline at end of file
diff --git a/third_party/libdrizzle/config.h b/third_party/libdrizzle/config.h
new file mode 120000
--- /dev/null
+++ b/third_party/libdrizzle/config.h
@@ -0,0 +1,1 @@
+config.h.linux
\ No newline at end of file
